src/transformers/modeling_tf_utils.py

```diff
@@ -1235,13 +1255,18 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
             # Load from a PyTorch checkpoint
             return load_pytorch_checkpoint_in_tf2_model(model, resolved_archive_file, allow_missing_keys=True)
 
-        model(model.dummy_inputs)  # build the network with dummy inputs
+        # we might need to extend the variable scope for composite models
+        if load_weight_prefix is not None:
+            with tf.compat.v1.variable_scope(load_weight_prefix):
```

Comment:
@jplu @sgugger @LysandreJik - this is the only way I found to make TFRag work with the current TF API. We need to hack the scope here because, depending on whether 

`TFRagModel.from_pretrained_encoder_generator` or `TFRagModel.from_pretrained` is used the scope of the models differs which without this hacky solution means that the pretrained weights are not loaded. 

This change does not affect any existing models

Reply:
If you are 100% sure that it doesn't affect all the other models' loading process, then I'm OK with it.

Reply:
Yeah, I'll run the whole TF SLOW test suite once the PR is 100% ready

Reply:
Let me know once you think the PR is 100% ready then. I will redo a complete review :)